mypy_django_plugin/transformers/manytomany.py

```diff
@@ -0,0 +1,144 @@
+    else:
+        # Avoid overwriting a decent 'through' argument
+        through_arg = default_return_type.args[1]
+
+    return default_return_type.copy_modified(args=[to_arg, through_arg])
+
+
+def get_m2m_arguments(
+    *,
+    ctx: FunctionContext,
+    model_info: TypeInfo,
+    django_context: DjangoContext,
+) -> Optional[M2MArguments]:
+    checker = helpers.get_typechecker_api(ctx)
+    to_arg = ctx.args[0][0]
+    to_model: Optional[ProperType]
+    if isinstance(to_arg, StrExpr) and to_arg.value == "self":
+        to_model = Instance(model_info, [])
+        to_self = True
+    else:
+        to_model = get_model_from_expression(to_arg, api=checker, django_context=django_context)
+        to_self = False
+
+    if to_model is None:
+        # 'ManyToManyField()' requires the 'to' argument
+        return None
+    to = M2MTo(arg=to_arg, model=to_model, self=to_self)
+
+    through = None
+    if len(ctx.args) > 5 and ctx.args[5]:
+        # 'ManyToManyField(..., through=)' was called
+        through_arg = ctx.args[5][0]
+        through_model = get_model_from_expression(through_arg, api=checker, django_context=django_context)
+        if through_model is not None:
+            through = M2MThrough(arg=through_arg, model=through_model)
+    elif not helpers.is_abstract_model(model_info):
+        # No explicit 'through' argument was provided and model is concrete. We need
+        # to dig up any generated through model for this 'ManyToManyField()' field
+        through_arg = None
+        m2m_throughs = helpers.get_django_metadata(model_info).get("m2m_throughs", {})
+        if m2m_throughs:
+            field_name = None
+            for defn in model_info.defn.defs.body:
+                if (
+                    isinstance(defn, AssignmentStmt)
+                    and defn.rvalue is ctx.context
+                    and len(defn.lvalues) == 1
+                    and isinstance(defn.lvalues[0], NameExpr)
+                ):
+                    field_name = defn.lvalues[0].name
+                    break
+
+            if field_name is not None:
+                through_model_fullname = m2m_throughs.get(field_name)
+                if through_model_fullname is not None:
+                    through_model_info = helpers.lookup_fully_qualified_typeinfo(checker, through_model_fullname)
+                    if through_model_info is not None:
+                        through = M2MThrough(arg=through_arg, model=Instance(through_model_info, []))
+
+    return M2MArguments(to=to, through=through)
+
+
+def get_model_from_expression(
+    expr: Expression,
+    *,
+    api: Union[TypeChecker, SemanticAnalyzer],
+    django_context: DjangoContext,
+) -> Optional[ProperType]:
+    """
+    Attempts to resolve an expression to a 'TypeInfo' instance. Any lazy reference
+    argument(e.g. "<app_label>.<object_name>") to a Django model is also attempted.
+    """
+    # TODO: Handle settings.AUTH_USER_MODEL?
```

Comment:
We have a bunch of these cases as well

Reply:
Something tells me this one will hit quite many projects.

Reply:
Guess it's fine to do in a separate PR, but maybe make sure to do it before the next release?

Reply:
Yeah, that's what I'm thinking too